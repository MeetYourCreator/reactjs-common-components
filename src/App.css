* {
  box-sizing: border-box;
  padding: 0px;
  margin: 0px;
  outline: none;
  text-decoration: none;
}

a {
  text-decoration: none;
}

input {
  width: 150px;
  padding: 1px 2px;
  border: 1px solid #43464b;
}

.appContainer {
  display: grid;
  grid-template-rows: 4fr 1fr;
  row-gap: 5px;
}

.screenContainer {
  width: 50vw;
  border-top: 3px solid black;
  border-right: 3px solid black;
}

.panels {
  grid-column-start: 0;
  grid-column-end: three;
}

.datepicker {
  display: flex;
  flex-direction: column;
  justify-items: center;
  align-items: center;
  grid-column-start: 3;
  grid-column-end: five;
}

/* .modal {
  display: grid;
  justify-items: center;
  align-items: center;
  grid-row-start: 2;
  grid-row-end: 3;
} */

.searchBar {
  display: grid;
  justify-items: center;
  grid-column-start: 0;
  grid-column-end: three;
}

.list {
  grid-column-start: 3;
  grid-column-end: five;
}
