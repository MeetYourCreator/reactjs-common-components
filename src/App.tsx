import { SplitViewPanels } from './components/';
import { Modal } from './components/';
import { SearchBar } from './components/';
import { DatePicker } from './components';
import { Toggle } from './components';
import { Input } from './components/';
import { Slider } from './components/';
import { InfiniteScroll } from './components/';
import { placeholderArray } from './utils';
import TestIMG from './assets/test/testimg.jpg';
import TestIMG2 from './assets/test/diner.jpg';
import './App.css';

export const App = () => {
  return (
    <>
      <section className="appContainer">
        <div className="screenContainer panels">
          <SplitViewPanels
            leftPanel={<div>Left Panel Content</div>}
            rightPanel={<div>Right Panel Content</div>}
          />
        </div>
        <div className="screenContainer datepicker">
          <DatePicker />
        </div>
        <div className="screenContainer modal">
          <Modal />
        </div>
        <div className="screenContainer searchBar">
          <SearchBar searchBarData={placeholderArray} />
        </div>
        <div className="screenContainer toggle">
          <Toggle />
        </div>
        <div className="screenContainer searchBar">
          <Input type="text" />
        </div>
        <div className="screenContainer">
          <Slider
            urlArray={[
              <Toggle />,
              <Input type="text" />,
              <img src={TestIMG} alt="testIMG" />,
              <img src={TestIMG2} alt="testIMG2" />,
            ]}
          />
        </div>
        <div className="screenContainer searchBar">
          <InfiniteScroll />
        </div>
      </section>
    </>
  );
};

export default App;
