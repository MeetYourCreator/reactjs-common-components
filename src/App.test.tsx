import React from 'react';
import { render, screen } from '@testing-library/react';
import App from './App';

test('renders SplitViewPanels component', () => {
  render(<App />);
  const divider = screen.getByRole('separator');
  expect(divider).toBeInTheDocument();
});
