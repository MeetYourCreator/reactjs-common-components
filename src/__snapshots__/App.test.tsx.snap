// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <section
    class="appContainer"
    id=":r0:"
  >
    <div
      class="screenContainer panels"
    >
      <section
        class="splitView "
      >
        <div>
          <div>
            Left Panel Content
          </div>
        </div>
        <div
          class="dividerHitbox"
        >
          <div
            class="divider"
          >
            <div>
              :::
            </div>
            <div>
              :::
            </div>
            <div>
              :::
            </div>
          </div>
        </div>
        <div
          class="rightPanel"
        >
          <div>
            Right Panel Content
          </div>
        </div>
      </section>
    </div>
    <div
      class="screenContainer panels"
    >
      <article
        class="vertical-nav-container"
      >
        <div
          class="card show"
        >
          <div
            class="cardContainer"
            id=":r1:"
            style="border: 10px solid green; color: white; width: 15rem;"
          >
            <header />
            <main
              class="cardContent"
            >
              <div
                class="cardImage"
              />
              <div
                class="cardText"
              >
                Card 0
              </div>
              <div
                class="cardChildren"
              >
                 
              </div>
            </main>
            <footer
              class="cardFooter"
            />
          </div>
        </div>
        <div
          class="card show"
        >
          <div
            class="cardContainer"
            id=":r2:"
            style="border: 10px solid yellow; color: white; width: 15rem;"
          >
            <header />
            <main
              class="cardContent"
            >
              <div
                class="cardImage"
              />
              <div
                class="cardText"
              >
                Card 1
              </div>
              <div
                class="cardChildren"
              >
                 
              </div>
            </main>
            <footer
              class="cardFooter"
            />
          </div>
        </div>
        <div
          class="card show"
        >
          <div
            class="cardContainer"
            id=":r3:"
            style="border: 10px solid blue; color: white; width: 15rem;"
          >
            <header />
            <main
              class="cardContent"
            >
              <div
                class="cardImage"
              />
              <div
                class="cardText"
              >
                Card 2
              </div>
              <div
                class="cardChildren"
              >
                 
              </div>
            </main>
            <footer
              class="cardFooter"
            />
          </div>
        </div>
        <div
          class="card show"
        >
          <div
            class="cardContainer"
            id=":r4:"
            style="border: 10px solid purple; color: white; width: 15rem;"
          >
            <header />
            <main
              class="cardContent"
            >
              <div
                class="cardImage"
              />
              <div
                class="cardText"
              >
                Card 3
              </div>
              <div
                class="cardChildren"
              >
                 
              </div>
            </main>
            <footer
              class="cardFooter"
            />
          </div>
        </div>
      </article>
    </div>
    <div
      class="screenContainer panels"
    >
      <section
        class="dialogContainer"
      >
        <button
          class="button"
          id=":r5:"
          style="position: absolute; z-index: 0;"
        >
          Open Modal
        </button>
      </section>
    </div>
    <div
      class="screenContainer panels"
    >
      <div
        class="datePickerContainer"
      >
        <form>
          <label
            for=":r6:"
          >
            selected-date
          </label>
          <input
            id=":r6:"
            name="selected-date"
            type="text"
            value=""
          />
        </form>
        <div
          class="dropdownContainer"
        />
      </div>
    </div>
    <div
      class="screenContainer panels"
    >
      <input
        class="toggle"
        id=":r7:"
        type="checkbox"
      />
      <label
        for=":r7:"
      />
    </div>
    <div
      class="screenContainer panels"
    >
      <form
        class="formContainer"
      >
        <label
          for=""
        />
        <input
          class="searchBar"
          type="select"
          value=""
        />
      </form>
    </div>
    <div
      class="screenContainer panels"
    >
      <div
        class="cardContainer"
        id=":r8:"
      >
        <header />
        <main
          class="cardContent"
        >
          <div
            class="cardImage"
          />
          <div
            class="cardText"
          />
          <div
            class="cardChildren"
          >
             
            <img
              src="diner.jpg"
            />
          </div>
        </main>
        <footer
          class="cardFooter"
        />
      </div>
    </div>
    <div
      class="screenContainer panels"
    >
      <article>
        <div
          class="element show"
        >
          <div
            class="cardContainer"
            id=":r9:"
          >
            <header />
            <main
              class="cardContent"
            >
              <div
                class="cardImage"
              />
              <div
                class="cardText"
              />
              <div
                class="cardChildren"
              >
                 R
              </div>
            </main>
            <footer
              class="cardFooter"
            />
          </div>
        </div>
      </article>
    </div>
  </section>
</DocumentFragment>
`;
